react/LocationSearch.tsx

```diff
@@ -111,7 +115,18 @@ const LocationSearch: React.FC<LocationSearchProps> = ({
     searchTerm,
     DEBOUNCE_DELAY_IN_MS
   )
-  const [suggestions, loadingSuggestions] = useSuggestions(debouncedSearchTerm)
+  const {
+    data: sessionTokenData,
+    loading: loadingSessionToken,
+    refetch: refetchSessionToken,
+  } = useQuery<Query, {}>(SESSION_TOKEN)
+  const sessionToken = loadingSessionToken
+    ? null
+    : sessionTokenData?.sessionToken ?? null
+  const [suggestions, loadingSuggestions] = useSuggestions(
+    debouncedSearchTerm,
+    sessionToken
+  )
```

Comment:
Maybe I'm overthinking here, but this is starting to get bloated. Does anyone have tips or suggestions on how to organize this?

Reply:
I was going to create an `useAddressAutocomplete` hook, similar to [Google's useAutocomplete](https://material-ui.com/pt/components/autocomplete/#customized-hook) or [this dude's usePlacesAutocomplete](https://dev.to/wellyshen/introducing-use-places-autocomplete-react-hook-for-google-maps-places-autocomplete-41h5), and although I would be able to hide at least the `sessionToken` from the component, the bloat would just be moved inside a hook (I guess). What do you ppl think?